packages/core/src/DatumBuilders/DatumBuilder.V3.class.ts

```diff
@@ -279,6 +280,26 @@ export class DatumBuilderV3 implements DatumBuilderAbstract {
     };
   }
 
+  private extractFeeManagerScript(
+    address?: string,
+  ): V3Types.TMultiSigScript | null {
+    if (!address) return null;
+
+    try {
+      const paymentHash =
+        Core.addressFromBech32(address).getProps().paymentPart?.hash;
+      if (!paymentHash) return null;
+
+      return { Address: { hex: paymentHash } };
+    } catch (error) {
+      console.warn(
+        `Failed to extract payment hash from feeManager address: ${address}`,
+        error,
+      );
+      return null;
+    }
```

Comment:
We should probably just throw an error here, because if the user passes in an invalid fee manager address, we don't want to fallback to anything.